mcs/class/corlib/System/TimeZoneInfo.cs

```diff
@@ -116,11 +116,11 @@ static TimeZoneInfo CreateLocal ()
 
 			try {
 				return FindSystemTimeZoneByFileName ("Local", "/etc/localtime");	
-			} catch {
+			} catch (TimeZoneNotFoundException) {
```

Comment:
It's different when the file is not found - we can search for alternatives because different systems use different paths (and no file is also a valid configuration).

A file present but containing invalid information - we should fail immediately (see e.g. http://martinfowler.com/ieeeSoftware/failFast.pdf).